### DISCUSSION TITLE
[Node Operator Question] Problems with deploying custom tracer (pre/post bedrock)

### DISCUSSION THREAD

**Discussion by kchojn** [ethereum-optimism/developers#230] [Node Operators]
### Are you running the most up to date node software?

- [X] Yes

### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I have a full archive node (https://github.com/smartcontracts/simple-optimism-node/blob/main/docker-compose.yml). 
I wanted to add a custom tracer (eth/tracers/native), this is how I add it to various chains on geth.

How to do this with Optimis? It should work on all blocks (pre/post bedrock). I tried to add it to op-geth, but it doesn't work, the logs show me, on l2geth, that it is undefined. So, can you let me know, how it should be deployed? Op-geth, l2geth, op-node? 

### Protocol Description

stable 

### Node Logs

```sh
curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"debug_traceBlockByNumber","params":["0x6ACFC00", {"tracer": "iris"}],"id":1}' xxx
```

```json
{"jsonrpc":"2.0","id":1,"result":[{"txHash":"0xc2407b946382bb4111f5e4ecda0aca8c22f0958f709930f13ca282c42233073d","error":"ReferenceError: iris is not defined at \u003ceval\u003e:1:2(0)"},{"txHash":"0x7a8cf6d38f25f1fc997f2f79c031f044e278f77e12a1c5dd3ab8bb26f79b7c5b","error":"ReferenceError: iris is not defined at \u003ceval\u003e:1:2(0)"},{"txHash":"0x0e8e85a5d40dd595ab058c2ea5349b7264e6f012c116dbcbe8e5d63fc9a19a41","error":"ReferenceError: iris is not defined at \u003ceval\u003e:1:2(0)"},{"txHash":"0x188857181023b5e4b70c364a595973746e8c1bab7e5f76de58f85d03e9efdee5","error":"ReferenceError: iris is not defined at \u003ceval\u003e:1:2(0)"},{"txHash":"0x224b16730284b3577aad9c91787dc46784123e3b17dbd24a308609158967f679","error":"ReferenceError: iris is not defined at \u003ceval\u003e:1:2(0)"},{"txHash":"0xe063376800915f346c16c1336ee471c4346e93c2f87083e7f5e0ceb7df35d5d3","error":"ReferenceError: iris is not defined at \u003ceval\u003e:1:2(0)"},{"txHash":"0x15c2823fe5385be2e746484b5ec7c2dd208d32e51faa86bf444c7359fd0fc56b","error":"ReferenceError: iris is not defined at \u003ceval\u003e:1:2(0)"},{"txHash":"0xab2da4ce78bd32965557cf3db47849ce411a775c0ad03fb30bb8094abf170cfa","error":"ReferenceError: iris is not defined at \u003ceval\u003e:1:2(0)"}]}
```

**Comment by kchojn** [comment]
Ok, i see that:
https://github.com/search?q=repo%3Aethereum-optimism%2Foptimism-audit-fixes+path%3A%2F%5El2geth%5C%2F%2F+callTracer&type=code

l2geth uses `*.js` tracers for pre-bedrock blocks? Can I deploy `.go` tracer to get output for pre-bedrock blocks? It works for post-bedrock in op-geth.

**Reply by sbvegan** [reply]
I'll see what I can find out about this one, but the changes from `l2geth` to `op-geth` are significant so my initial assumption is you won't be able to do this for pre-bedrock blocks. Legacy geth (`l2geth`) is also using an older version of `geth` as a base and I don't think tracing was as robust at the time.
